cps/db.py

```diff
@@ -957,7 +957,18 @@ def get_cc_columns(self, config, filter_config_custom_read=False):
     def get_search_results(self, term, config, offset=None, order=None, limit=None, *join):
         order = order[0] if order else [Books.sort]
         pagination = None
-        result = self.search_query(term, config, *join).order_by(*order).all()
+        
+        # search also through the subtitles (for videos)
+        other_terms = lb_search.get_search_terms(term)
+        # lb_search.get_search_terms returns a list of video titles, "term" parameter is expected to be a book/video title
+        term = [term] + other_terms
+
+        result = list()
+        for term_part in term:
+            # the search_query function below only searches for books titles
+            result += self.search_query(term_part, config, *join).order_by(*order).all()
+        # we need to remove duplicates because the same book/video could be found multiple times
+        result = list(set(result))
```

Comment:
I'm asking myself if (a) user's **search term/query** (e.g. "feelings") and (b) **lists of video/book titles whose subtitles contain that term/query** ...can be more crisply/cleanly disambiguated.

@deldesir can you please clarify:

- Is var `result` on Line 971 of cps/db.py (in PR #152) a list of Calibre-Web book/video IDs — e.g. Calibre-Web's actual counting numbers like [1, 2, 3, 27] that appear in its web UI?  i.e. Is that what function `search_query` outputs, do you know?  https://github.com/iiab/calibre-web/blob/a5486db89c4db127152a02815bc59d2e76ed2a31/cps/db.py#L909-L938
- Or, maybe it's list of some equivalent book/video pointers within SQLite ?

_(Please paste in an actual `result` sample, as an example will be extremely useful!)_

Reply:
@deldesir are all these variables that start with `cc` can `cc_` about Calibre and/or Calibre-Web ["custom columns"](https://manual.calibre-ebook.com/template_lang.html#using-templates-to-define-custom-columns) ?

(In function `search_query` and similar functions, within `cps/db.py` ?)

And if so, can we mostly ignore those for now?!

Reply:
The `(7)` in the above output definitely appears to be the book/video ID, and other {tags, series, authors, publishers} (ETC!) aspects of the `result` variable might be more understandable with:

https://github.com/iiab/calibre-web/blob/a5486db89c4db127152a02815bc59d2e76ed2a31/cps/db.py#L363-L390